### PR TITLE
Toggle mediaelement fullscreen property on doublelick.

### LINKED ISSUE #442: Toggle fullscreen on double click
Its a common UX feature for video to toggle the fullscreen mode when double clicking on the player. I have tripped over this many times on media.ccc.de so it would be fantastic if this could be added. Thanks already for this great platform! :)

### PR DESCRIPTION
**Pull request by Foaly**
This is supposed to be a fix for #442.
Since I don't have a test environment, I was not able to test this. I just took an educated guess on where the change should go. I hope this doesn't cause more work than it should! :) 

(closes #442)

### CODE REVIEW

**Comment by saerdnaer** [comment]
The problem with that implementation is, that I actually wanted to implement a "skip back/ skip back via double touch on the left/right 15% of the screen width" feature, which somehow overlaps with this... lets see if I find a solution to implement both.

**Comment by Necr0** [comment]
I think both could be implemented by having the skip forward/backward feature listen to `touchstart`/`touchend` events exclusively and preventing the `click` event thus not causing a `dblclick` event. Since there is no dbltouch event or something like that that would have to be implemented which shouldn't be that big of a problem.

**Comment by Foaly** [comment]
Ah I see!
@saerdnaer Could you maybe clarify if you meant to have the screen edge area react to `touch` events OR to `click` events (or both)? 
I think for `touch` events the solution @Necr0 proposed would work quite nice. Because I misunderstood I already started thinking about making both fullscreen and skip possible for `click` events. But I'm not sure how usefull this would be.

**Comment by saerdnaer** [comment]
I tested the PR locally, but it did not work as expected – the player is probably swallowing all click events. 

To get to a local test setup under macOS (without docker):

* Download and install https://postgresapp.com
* Install ruby 2.6.x as described at https://gorails.com/setup/osx/

```
git clone git@github.com:voc/voctoweb.git
cd voctoweb
bundle install
./bin/setup
rake db:migrate
export SKIP_ELASTICSEARCH=1
./bin/update-data
rails server
```

**Comment by saerdnaer** [comment]
BTW: I also exposed `player` and `mediaElement` in the player page as global variables. You can go to any video on media.ccc.de, open the javascript console of your browser and paste
```javascript
mediaElement.addEventListener('dblclick', function() {
   player.fullscreen();
});
```
and test this change yourself.

File: app/views/frontend/shared/_player_video.haml [hunk 1]
```diff
@@ -80,6 +80,10 @@
                 console.log(event);
             }
           });
+
+          mediaElement.addEventListener('dblclick', function() {
+            player.fullscreen();
```
Comment (saerdnaer) [inline app/views/frontend/shared/_player_video.haml]: This event listener does not work, please find out why and fix it :-)

**Comment by Foaly** [comment]
I did a few more small test and it seems that something is overlaying the mediaelement while its paused (I'm guessing the overlay with the voctocat).
But I also noticed that my javascript skills are way to rusty to debug deeper, so if somebody else would take this over I would be glad :)

**Comment by Foaly** [comment]
I've been watching some media.ccc.de recently and I stumbled upon my habit again to double click a video to make it go fullscreen. Then I rediscovered this issue again. Maybe we can get it unstuck from limbo 🙃 

I have investigated a little bit further. My conclusion from last year that the voctocat overlay blocks the player seems to be incorrect. There seems to be another problem. When I select the `mediaElement` either in the console or using the element picker it only select a tiny slice of the "player" at the bottom edge. See this picture: 
Bildschirmfoto 2021-01-08 um 20 54 40
So if an event handler is attached to this element, there is only a tiny space to receive the clicks, which I assume is why the previous attempt did not work.

If I select the video inside the media element, then the whole "player" element is selected. See this picture:
Bildschirmfoto 2021-01-08 um 20 54 59

If I try using the following code in the console I can get the desired double click behaviour:
``` javascript
mediaElement.firstChild.addEventListener('dblclick', function() {
   player.enterFullScreen()
});
```

It works, but it seems quite hacky to me though... Since I am quite unfamiliar with the voctoweb architecture I was wondering, where are the other interactions (like play/pause on single click) defined? Maybe that would be the more appropriate place for this change!

Thanks for all you great work so far :)

**Pull request closed without merging**